### ISSUE TITLE
ASM submodule not exposing `use_tf_google_credentials_env_var` to forward to kubectl-wrapper submodule

### ISSUE THREAD

**Issue by torbendury** [terraform-google-modules/terraform-google-kubernetes-engine#1443]
### TL;DR

When trying to deploy ASM on our GKE, the module calls the `kubectl-wrapper` to deploy everything. The ASM submodule is not exposing the variable `use_tf_google_credentials_env_var` to the `kubectl-wrapper` submodule which makes it hard to use it.

### Expected behavior

I expect the `asm` module to expose the variable `use_tf_google_credentials_env_var` which can be set to `true`. It should forward this variable to the [internally called](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/c81c2fec943df1d891495b2afe89b9904203a585/modules/asm/main.tf#L56) `kubectl-wrapper`.

This way, we can run our CI/CD without having to run `gcloud auth [...]` before, but rather can rely on the `GOOGLE_APPLICATION_CREDENTIALS` - or in this case: `GOOGLE_CREDENTIALS`.

### Observed behavior

The `asm` module does not expose the variable, so we would need to run `gcloud auth` beforehand every time we run our CI/CD pipelines.

The reason this would need to happen everytime is that we use containerized CI/CD environments that are not authenticated by default but rely on the `GOOGLE_APPLICATIONS_CREDENTIALS` working.

### Terraform Configuration

```hcl
module "asm" {
  source                    = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  version                   = "23.2.0"
  project_id                = local.project_name
  cluster_name              = google_container_cluster.XXXXXX.name
  cluster_location          = "europe-west1"
  multicluster_mode         = "connected"
  enable_cni                = true
  enable_mesh_feature       = true
  enable_fleet_registration = true
  # not forwarded to the internal submodule
  # use_tf_google_credentials_env_var = true
}
```

### Terraform Version

```sh
Terraform v1.3.3
+ provider registry.terraform.io/hashicorp/google v4.37.0
+ provider registry.terraform.io/hashicorp/google-beta v4.37.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.14.0
```

### Additional information

I think adding the variable to the module is not going to break anything if it is set to `false` by default (which would replicate the state that now happens) and stays optional. I don't think that the change is going to be very big.

**Comment by frits-v** [comment]
@morgante this is still an issue folks are facing. Current work around in most environments isn't pretty (e.g. pre-plan / apply hooks to populate the credentials).

**Comment by torbendury** [comment]
Has this been fixed? I don't see it anywhere, so it is still a problem which required ugly workarounds..
